### PR TITLE
FromToPicker: add internationalization and typescript

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Hopefully these will get easier from here on out as the process gets more familiar.

Some notes:
- I haven't used decorators in the stories (although they could be used to make the Intl stuff easier) since I thought that would be overkill for such a small component
- I'm a bit upset about all the `flatten()` stuff because it feels lousy to have to include this hack in every package individually. Hope there's a better solution... Maybe something in core-utils?
- This package fails to load without an `<IntlProvider>`. Are we ok with this?
- The `messages.ts` feels like a hack that's annoying to have to replicate across every package. I hope that someone has some good suggestions for improving this!
- I'm a little unhappy with how we're providing the custom messages. Hoping someone has a better alternative.

### CODE REVIEW

File: packages/from-to-location-picker/src/index.story.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export const smallTextSansSerif = (): React.ReactElement => (
+  <IntlProvider locale="en-US">
+    <span style={{ fontSize: "75%", fontFamily: "sans-serif" }}>
+      Plan a trip:
+      <FromToLocationPicker onFromClick={onFromClick} onToClick={onToClick} />
+    </span>
+  </IntlProvider>
+);
+
+export const otpAmbientStyle = (): React.ReactElement => (
```
Comment (binh-dam-ibigroup) [inline packages/from-to-location-picker/src/index.story.tsx:32]: Rename this story to say "custom style and text".

File: packages/from-to-location-picker/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+export type Location = {
+  lat: number;
+  lon: number;
+  name: string;
+  /**
+   * This is only used location that a user has saved. Can be either:
```
Comment (binh-dam-ibigroup) [inline packages/from-to-location-picker/src/types.ts:6]: First sentence does not make sense. Do you mean something like "This is a user-defined location type"?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Looks mostly good, see the minor changes suggested.

File: packages/from-to-location-picker/src/index.story.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+export default {
+  title: "From-To-Picker",
+  component: FromToLocationPicker
+};
+
+const onFromClick = action("onFromClick");
+const onToClick = action("onToClick");
+
+export const fromTo = (): React.ReactElement => (
+  <IntlProvider locale="en-US">
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/from-to-location-picker/src/index.story.tsx:18]: I've done this, but I'm weary of committing it as it seems to make the snapshots a bit useless. Screen Shot 2022-01-18 at 7 39 57 AM

File: packages/from-to-location-picker/src/types.ts (hunk 2 above)
Comment (miles-grant-ibigroup) [reply packages/from-to-location-picker/src/types.ts]: I copy pasted this from the flow types, but I agree something is wrong here. I've corrected it making as few changes as possible in 57e3522f13c3e0c11b4eef871ff4200451264d54

File: packages/from-to-location-picker/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      </S.LocationPickerSpan>
+      <S.LocationPickerSpan>
+        {showIcons && <LocationIcon type="to" size={iconSize} />}
+        <S.Button onClick={handleToClick}>
+          <FormattedMessage
+            defaultMessage={defaultMessages["otpUi.FromToLocationPicker.to"]}
+            description='Text for the "to" button of the picker'
+            id="otpUi.FromToLocationPicker.to"
+          />
+    </S.Button>
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/from-to-location-picker/src/index.tsx]: Prettier run in 57e3522f13c3e0c11b4eef871ff4200451264d54

File: packages/from-to-location-picker/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+          id="otpUi.FromToLocationPicker.from"
+        />
+        </S.Button>
+      </S.LocationPickerSpan>
+      <S.LocationPickerSpan>
+        {showIcons && <LocationIcon type="to" size={iconSize} />}
+        <S.Button onClick={handleToClick}>
+          <FormattedMessage
+            defaultMessage={defaultMessages["otpUi.FromToLocationPicker.to"]}
+            description='Text for the "to" button of the picker'
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/from-to-location-picker/src/index.tsx]: Done in 57e3522f13c3e0c11b4eef871ff4200451264d54

File: packages/from-to-location-picker/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  };
+
+  return (
+    <S.FromToPickerSpan>
+      <S.LocationPickerSpan>
+        {showIcons && <LocationIcon type="from" size={iconSize} />}
+        <S.Button onClick={handleFromClick}>
+        <FormattedMessage
+          defaultMessage={defaultMessages["otpUi.FromToLocationPicker.from"]}
+          description='Text for the "from" button of the picker'
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/from-to-location-picker/src/index.tsx]: Done in 57e3522f13c3e0c11b4eef871ff4200451264d54

File: packages/from-to-location-picker/src/index.story.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/from-to-location-picker/src/index.story.tsx]: Good call! Was wondering what ambient style meant. Renamed in 57e3522f13c3e0c11b4eef871ff4200451264d54

File: packages/from-to-location-picker/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
+import flatten from "flat";
+import LocationIcon from "@opentripplanner/location-icon";
+import { FormattedMessage } from "react-intl";
+import React from "react";
+// eslint-disable-next-line prettier/prettier
+import type { FromToPickerProps } from "./types";
```
Comment (binh-dam-ibigroup) [inline packages/from-to-location-picker/src/index.tsx:6]: Move `import type` between `import styled` and `import defaultEnglishMessages`.

**Review by binh-dam-ibigroup — APPROVED** [review]
Thanks for the changes. One more tiny thing, but o/w looks good to me!

File: packages/from-to-location-picker/src/index.tsx (hunk 7 above)
Comment (miles-grant-ibigroup) [reply packages/from-to-location-picker/src/index.tsx]: Good note thank you! Reordered in da90c95eb425b586d677d17eb3885b216bb45271

**Review by demory — APPROVED** [review]

**Pull request merged**